google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java

```diff
@@ -1362,7 +1366,7 @@ private static QueryResponse getQueryResults(
                     : jobId.getLocation());
     try {
       GetQueryResultsResponse results =
-          runWithRetries(
+          BigQueryRetryHelper.runWithRetries(
```

Comment:
Can we add this to also `queryRpc` (line 1278). That should also allow you to verify idempotency using `requestId`.

Reply:
Sure, I have added `BigQueryRetryHelper.runWithRetries` for ` TableResult query(...)` and have implemented testcase `testFastQueryRateLimitIdempotency` to test the idempotency

Reply:
Great - thank you.

Reply:
Sure, I have implemented `BigQueryRetryHelper.runWithRetries` on `QueryResponse waitForQueryResults` method, which is used by `TableResult getQueryResults` method